python/cudf/cudf/core/index.py

```diff
@@ -1478,30 +1478,31 @@ class RangeIndex(Index):
     Examples
     --------
     >>> import cudf
-    >>> cudf.RangeIndex(0, 10, name="a")
-    RangeIndex(start=0, stop=10, name='a')
+    >>> cudf.RangeIndex(0, 10, 1, name="a")
+    RangeIndex(start=0, stop=10, step=1, name='a')
 
-    >>> cudf.RangeIndex(range(1, 10), name="a")
-    RangeIndex(start=1, stop=10, name='a')
+    >>> cudf.RangeIndex(range(1, 10, 1), name="a")
+    RangeIndex(start=1, stop=10, step=1, name='a')
     """
 
     def __new__(
-        cls, start, stop=None, step=None, dtype=None, copy=False, name=None
+        cls, start, stop=None, step=1, dtype=None, copy=False, name=None
     ) -> "RangeIndex":
 
-        if step is not None:
-            raise NotImplementedError("step is not yet supported")
+        if step == 0:
+            raise ValueError("Step must not be zero.")
 
         out = Frame.__new__(cls)
         if isinstance(start, range):
             therange = start
             start = therange.start
             stop = therange.stop
+            step = therange.step
         if stop is None:
             start, stop = 0, start
         out._start = int(start)
         out._stop = int(stop)
-        out._cached_values = None
+        out._step = int(step)
```

Comment:
```suggestion
        out._step = int(step) if step is not None else 1
```

We should also be handling the case when `step` is passed as `None`.